### PR TITLE
Feat/Allow admin to move workplace

### PR DESCRIPTION
**Pull request by ross-mcfadyen**
#### Tests
Does this PR include tests for the changes introduced?
- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: server/test/unit/routes/admin/move-workplace/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+const sinon_sandbox = sinon.createSandbox();
+
+const models = require('../../../../../models/index');
+const { moveWorkplaceAdmin } = require('../../../../../routes/admin/move-workplace');
+
+const { establishmentBuilder } = require('../../../../factories/models');
+
+const { cloneDeep } = require('lodash');
+
+describe('server/routes/admin/moveWorkplaceAdmin.js', () => {
```
Comment (chadtomkiss-madetech) [inline server/test/unit/routes/admin/move-workplace/index.spec.js:13]: Given the introduction of `celebrate` should we have tests around what happens if you submit a non UUID value to the endpoint?

File: src/app/features/workplace/workplace-info-panel/workplace-info-panel.component.ts [hunk 2]
```diff
@@ -134,4 +137,17 @@ export class WorkplaceInfoPanelComponent implements OnInit, OnDestroy {
   ngOnDestroy(): void {
     this.subscriptions.unsubscribe();
   }
+
+  public isLoggedIn(): boolean {
+    return this.authService.isAuthenticated();
+  }
+
+  public isAdminUser(): boolean {
+    return this.authService.isAdmin;
```
Comment (chadtomkiss-madetech) [inline src/app/features/workplace/workplace-info-panel/workplace-info-panel.component.ts:146]: This function is duplicated in a few components.. any way for us to re-use it?

**Review by chadtomkiss-madetech — APPROVED** [review]

File: src/app/features/workplace/workplace-info-panel/workplace-info-panel.component.ts (hunk 2 above)
Comment (chadtomkiss-madetech) [reply src/app/features/workplace/workplace-info-panel/workplace-info-panel.component.ts:146]: If you make the `authService` public, you could do:

```
*ngIf="authService.isAuthenticated() && authService.isAdmin"
```

File: server/routes/admin/move-workplace/index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    res.status(200).send();
+  } catch (error) {
+    console.error(error);
+
+    res.status(503).send();
+  }
+};
+
+router.route('/').post(
+  celebrate({
```
[the remark this answers is not in the document]
Comment (ross-mcfadyen) [reply server/routes/admin/move-workplace/index.js:32]: Haha, yeah it's pretty cool but couldn't work out how to unit test it. We decided to push on - maybe it's a tech debt thing I could look at... The library works, but whenever we tried it through the testing suite we couldn't get the validations to return 🤷

[the remark this answers is not in the document]
Comment (ross-mcfadyen) [reply server/routes/admin/move-workplace/index.js:32]: If you've an example of it being tested somewhere, I'd love to see it.

File: server/test/unit/routes/admin/move-workplace/index.spec.js (hunk 1 above)
Comment (ross-mcfadyen) [reply server/test/unit/routes/admin/move-workplace/index.spec.js:13]: Ah. See above....

Comment (ross-mcfadyen) [reply server/test/unit/routes/admin/move-workplace/index.spec.js:13]: I agree, yes we should, we just hit a bit of a dead end.

File: src/app/features/workplace/workplace-info-panel/workplace-info-panel.component.ts (hunk 2 above)
Comment (ross-mcfadyen) [reply src/app/features/workplace/workplace-info-panel/workplace-info-panel.component.ts:146]: Sure, I'll take a look.

Comment (ross-mcfadyen) [reply src/app/features/workplace/workplace-info-panel/workplace-info-panel.component.ts:146]: Update - I'd like to do this, but I'll do it in a subsequent PR. This has been hanging around for a while and I don't think it should block the merge.

File: src/app/shared/components/link-to-parent/link-to-parent-dialog.component.ts [hunk 4]
```diff
@@ -171,7 +171,7 @@ export class LinkToParentDialogComponent extends DialogComponent implements OnIn
   public sendRequestToParent() {
     if (this.form.valid) {
       this.permissionType = this.form.value.dataPermission;
-      this.parentWorkplaceId = this.getParentUidOrName(this.form.value.parentNameOrPostCode, 'uid') || null;
+      this.parentWorkplaceId = this.getParentUidOrName(this.form.value.parentNameOrPostCode, 'uid') || null; //
```
Comment (chadtomkiss-madetech) [inline src/app/shared/components/link-to-parent/link-to-parent-dialog.component.ts:174]: Looks like a hanging comment `//`

File: src/app/shared/components/move-workplace/move-workplace-dialog.component.ts [hunk 5]
```diff
@@ -149,7 +149,35 @@ export class MoveWorkplaceDialogComponent extends DialogComponent implements OnI
 
   private moveWorkplaceAdmin(): void {
     // Do stuff
-    this.parentWorkplaceId = this.getParentUidOrName(this.form.value.parentNameOrPostCode, 'uid') || null;
+    console.log('whoops');
```
Comment (chadtomkiss-madetech) [inline src/app/shared/components/move-workplace/move-workplace-dialog.component.ts]: Can remove the `console.log` I think?

File: src/app/shared/components/move-workplace/move-workplace-dialog.component.ts [hunk 6]
```diff
@@ -149,7 +149,35 @@ export class MoveWorkplaceDialogComponent extends DialogComponent implements OnI
...
+
+    if (this.form.valid) {
```
Comment (chadtomkiss-madetech) [inline src/app/shared/components/move-workplace/move-workplace-dialog.component.ts:151]: It looks like the `moveWorkplaceAdmin` function doesn't run if the form is not `valid`. Do you think we still need the `if`?

If so, is it ok to continue on with a `null` `this.parentWorkplaceId`?

**Review by chadtomkiss-madetech — APPROVED** [review]

**Pull request merged**